### PR TITLE
DRAFT chore(edit-ema): POC for edit page use new edit ema

### LINKED ISSUE #27427: Edit EMA: POC on how to use it with vtl pages
### Parent Issue

#26644 

### Task

POC to edit VTL rendered pages with new editor

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

_No response_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by fmontes**
### Proposed Changes
* Test how to render the HTML
* Add event listeners

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts [hunk 1]
```diff
@@ -153,6 +155,13 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
         this.store.updateEditorState(EDITOR_STATE.LOADING);
     }
 
+    ngAfterViewInit(): void {
+        const doc = this.iframe.nativeElement.contentWindow?.document;
+        doc.open();
+        doc.write(PAGE_RENDER);
+        doc.close();
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts:162]: Write the HTML that comes from the Page API.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/utils/fake-page.ts [hunk 2]
```diff
@@ -0,0 +1,1899 @@
...
+            <div class="col-lg-12 banner-tall">
+              <div
+                data-dot-object="vtl-file"
+                data-dot-inode="ef208def-a562-47a5-ac46-958e84feb3a9"
+                data-dot-url="preloop.vtl"
+                data-dot-can-read="true"
+                data-dot-can-edit="true"></div>
+              <div class="container-notes">Banner</div>
+              <div
+                data-dot="container"
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/utils/fake-page.ts:282]: Now we need data-dot and data-content, that's it.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/utils/fake-page.ts [hunk 3]
```diff
@@ -0,0 +1,1899 @@
...
+            action: 'set-contentlet',
+            payload: {
+                x,
+                y,
+                width,
+                height,
+                payload: contentletPayload
+            }
+        }, '*');
+      });
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/utils/fake-page.ts:1895]: And we inject the JavaScript to emit the events... now the tricky part is the most of this code will be the same as we do in react, we need to find a way to share the code so we don't have to write and test it twice.

Comment (zJaaal) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/utils/fake-page.ts:1895]: We can try to make it as reusable as possible and move it to the `client`

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts (hunk 1 above)
Comment (fmontes) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts:162]: I found a better way to do this, using blob

**Pull request closed without merging**